### PR TITLE
Abstract listener, read and write only clients. For review only.

### PR DESCRIPTION
**Pull request by vrozov**

### CODE REVIEW

File: src/main/java/com/datatorrent/netlet/AbstractClient.java [hunk 1]
```diff
@@ -184,40 +121,27 @@ else if (read == -1) {
   /**
    * @since 1.2.0
    */
+  @Override
```
Comment (243826) [inline src/main/java/com/datatorrent/netlet/AbstractClient.java]: delete this method?

**Comment by 243826** [comment]
why is it not just a property instead of separate classes?

**Comment by vrozov** [comment]
New classes are introduced to keep binary compatibility for existing classes and to minimize the complexity of dealing with reads in write-only classes and with writes in read-only classes.

File: src/main/java/com/datatorrent/netlet/AbstractClient.java (hunk 1 above)
Comment (vrozov) [inline src/main/java/com/datatorrent/netlet/AbstractClient.java]: The goal of keeping this method is to preserve @since tag.

**Pull request closed without merging**